compiler/rustc_mir_transform/src/inline.rs

```diff
@@ -770,6 +770,10 @@ fn check_codegen_attributes<'tcx, I: Inliner<'tcx>>(
         return Err("never inline attribute");
     }
 
+    if let OptimizeAttr::DoNotOptimize = callee_attrs.optimize {
+        return Err("no optimize attribute");
```

Comment:
This sounds like the error is that the function does not have an optimize attribute, not that it has an attribute that says to not optimize. Perhaps "has DoNotOptimize attribute"? Or such?